tensorflow/compiler/tf2xla/xla_op_registry.cc

```diff
@@ -174,6 +175,26 @@ XlaOpRegistry::~XlaOpRegistry() = default;
   }();
   (void)registration_init;
 
+  // Register GPU JIT devices for NextPluggableDevice if its jit_device_type is
+  // `XLA_GPU_JIT`.
+  if (DeviceFactory::IsPluggableDevice(device_name) &&
+      registry.compilation_devices_.find(device_name) ==
+          registry.compilation_devices_.end()) {
```

Comment:
`registry.compilation_devices_.find(device_name) == registry.compilation_devices_.end()` requires `lock(registry.mutex_)` as well. maybe move it inside `if (DeviceFactory::IsPluggableDevice(device_name))`?

Reply:
done